tools/upgrade_versions.py

```diff
@@ -224,13 +224,24 @@ def run(CONF):
     committer_name = CONF.get('default').get('committer_name')
     committer_email = CONF.get('default').get('committer_email')
 
-    REQUIRED_PARAMETERS = ["push_repo_url", "push_repo_branch",
-                           "committer_name", "committer_email"]
+    REQUIRED_PARAMETERS = ["committer_name", "committer_email"]
+                           
+    REQUIRED_PUSH_REPO_PARAMETERS = ["push_repo_url", "push_repo_branch"]
+
     for parameter in REQUIRED_PARAMETERS:
         if CONF.get('default').get(parameter) is None:
             LOG.error("Parameter '%s' is required", parameter)
             return 1
 
+    push_repo = True
```

Comment:
What if we got this boolean from the config? This way, the user would have to explicitly disable pushing (e.g. by passing --local-update) and we could just concatenate the list `PUSH_REPO_PARAMETERS` into `REQUIRED_PARAMETERS` if it's `True`, removing the second for.